content/src/view.js

```diff
@@ -1350,7 +1351,47 @@ $(window).on('resize.listing', function() {
   }
 });
 
-function updatePaneLinks(pane) {
+
+var searchCache = "file-search";
+var prevSearch;
+function updateSearchResults(search,pane) {
+  search = search.toLowerCase();
+  if (search == prevSearch){
+    return;
+  }
+    
+  if (!state.pane[pane].allLinks){
+    state.pane[pane].allLinks = state.pane[pane].links;
+  }
+    
+  var results;
+  if (search) {
+    var savedCache=cache.get(searchCache,search);
+    if (savedCache) {
+      results = savedCache;
+    } else {
+      results = [];
+      var list = state.pane[pane].allLinks;
+
+      for (j = 0; j < list.length; j++) {
+        if (list[j].name.toLowerCase().indexOf(search) == 0) {
```

Comment:
So, what you are saying is to do the searching in the server when the search result is already there in the client cache.

Yes, It's correct. since there is a limit of 600.